src/jdk.incubator.jextract/share/classes/jdk/internal/jextract/impl/OutputFactory.java

```diff
@@ -361,6 +361,14 @@ public Void visitFunction(Declaration.Function funcTree, Declaration parent) {
                 default:
                     return getAsFunctionPointer(((Type.Delegated) type).type());
             }
+        } else if (type instanceof Type.Function) {
```

Comment:
I think the check before this can be simplified. Essentially, in this routine we should have three cases:

* we have found a function type -> return it
* we have found a delegate type -> recurse on the delegatee
* none of the above, return null